modules/landing-pages/module.php

```diff
@@ -0,0 +1,342 @@
+		if ( $this->trashed_posts ) {
+			return $this->trashed_posts;
+		}
+
+		$this->trashed_posts = new \WP_Query( [
+			'post_type' => 'page',
+			'post_status' => 'trash',
+			'elementor_library_type' => self::DOCUMENT_TYPE,
+			'meta_key' => '_elementor_template_type',
+			'meta_value' => self::DOCUMENT_TYPE,
+		] );
+
+		return $this->trashed_posts;
+	}
+
+	private function get_landing_page_posts() {
+		if ( $this->posts ) {
+			return $this->posts;
+		}
+
+		$this->posts = new \WP_Query( [
+			'post_type' => 'page',
+			'elementor_library_type' => self::DOCUMENT_TYPE,
+			'meta_key' => '_elementor_template_type',
+			'meta_value' => self::DOCUMENT_TYPE,
+		] );
+
+		return $this->posts;
+	}
+
+	/**
+	 * Is Elementor Landing Page.
+	 *
+	 * Check whether the post is an Elementor Landing Page.
+	 *
+	 * @since 3.1.0
+	 * @access public
+	 *
+	 * @param \WP_Post $post Post Object
+	 *
+	 * @return bool Whether the post was built with Elementor.
+	 */
+	public function is_elementor_landing_page( $post ) {
+		// If the post is not a page, save a call to the DB.
+		if ( 'page' !== $post->post_type ) {
+			return false;
+		}
+
+		return 'landing-page' === get_post_meta( $post->ID, Document::TYPE_META_KEY, true );
+	}
+
+	/**
+	 * Add Submenu Page
+	 *
+	 * Adds the 'Landing Pages' submenu item to the 'Templates' menu item.
+	 *
+	 * @since 3.1.0
+	 */
+	private function add_submenu_page() {
+		$posts = $this->get_landing_page_posts();
+
+		// If there are no Landing Pages, show the "Create Your First Landing Page" page.
+		// If there are, show the pages table.
+		if ( ! empty( $posts->posts ) ) {
+			$this->landing_page_menu_slug = self::ADMIN_PAGE_SLUG;
+			$landing_page_menu_callback = null;
+		} else {
+			$this->landing_page_menu_slug = self::DOCUMENT_TYPE;
+			$landing_page_menu_callback = [ $this, 'print_empty_landing_pages_page' ];
+		}
+
+		$landing_pages_title = __( 'Landing Pages', 'elementor' );
+
+		add_submenu_page(
+			Source_Local::ADMIN_MENU_SLUG,
+			$landing_pages_title,
+			$landing_pages_title,
+			'manage_options',
+			$this->landing_page_menu_slug,
+			$landing_page_menu_callback
+		);
+	}
+
+	private function get_add_new_landing_page_url() {
+		return Utils::get_create_new_post_url( 'page', self::DOCUMENT_TYPE ) . '#library';
+	}
+
+	/**
+	 * Remove Type Column
+	 *
+	 * Removes the "Type" column from the pages table.
+	 *
+	 * @since 3.1.0
+	 *
+	 * @param array $columns
+	 * @return array $columns
+	 */
+	private function remove_type_column( $columns ) {
+		unset( $columns['taxonomy-elementor_library_type'] );
+
+		return $columns;
+	}
+
+	/**
+	 * Get Empty Landing Pages Page
+	 *
+	 * Prints the HTML content of the page that is displayed when there are no existing landing pages in the DB.
+	 * Added as the callback to add_submenu_page.
+	 *
+	 * @since 3.1.0
+	 */
+	public function print_empty_landing_pages_page() {
+		$template_sources = Plugin::$instance->templates_manager->get_registered_sources();
```

Comment:
Maybe it can work with `maybe_render_blank_state`?

Reply:
I used a portion of that method which I extracted.